lib/learn_test/runner.rb

```diff
@@ -0,0 +1,65 @@
+module LearnTest
+  class Runner
+    attr_reader :repo, :options
+    SERVICE_URL = 'http://ironbroker.flatironschool.com'
+
+    def initialize(repo, options = {})
+      @repo = repo
+      @options = options
+      die if strategies.empty?
+    end
+
+    def run
+      strategies.each do |strategy|
+        strategy.check_dependencies
+        strategy.configure
+        strategy.run
+        if !help_option_present? && strategy.push_results?
+          push_results(strategy)
+          strategy.cleanup unless keep_results?
+        end
+      end
+    end
+
+    def files
+      @files ||= Dir.entries('.')
+    end
+
+    def keep_results?
+      @keep_results ||= options[:keep] || !!options.delete('--keep')
+    end
+
+    private
+
+    def connection
+      @connection ||= Faraday.new(url: SERVICE_URL) do |faraday|
+        faraday.adapter  Faraday.default_adapter
+      end
+    end
+
+    def strategies
+      @strategies ||= LearnTest::Strategies.constants.map{ |s| LearnTest::Strategies.const_get(s).new(self) }.select{ |s| s.detect }
```

Comment:
i might remove this fancy metaprogramming stuff and save it for a future release because what if it detects two strategies in the same repo, i think we'd wanna have the strategies in an array so we can detect the first one from the list in order of priority